more-info-climate.html use supported_features

ciotlosm opened the pull request:
- Make more-info-climate.html `supported_features` to display controls
- Fix bug with Fan Mode not showing selection https://github.com/home-assistant/home-assistant-polymer/issues/852
- `temperature_high` and `temperature_low` fixes - improves when hides/shows

Update: Demo platform now fixed as part of https://github.com/home-assistant/home-assistant/pull/12070

ciotlosm committed:
Added supported features for controls

ciotlosm committed:
Improved supported features

ciotlosm committed:
Removed lines resulted from rebase

ciotlosm commented:
@andrey-git Please have a look and let me know if using supported_features is feasible. Maybe this is too big breaking change.

andrey-git commented on src/more-infos/more-info-climate.html:
```diff
@@ -322,19 +323,18 @@
-    return stateObj.attributes.target_temp_low ||
-        stateObj.attributes.target_temp_high;
+  supportsTemperatureRange(stateObj) {
+    return ((stateObj.attributes.supported_features & 2) !== 0)
```
return (stateObj.attributes.supported_features & 4) !== 0

ciotlosm replied:
Will fix that next, but I'm still unsure if using supported_features is correct because it breaks all climate demo components display.

ciotlosm committed:
Fixed lint and review

andrey-git commented on src/more-infos/more-info-climate.html:
```diff
@@ -322,19 +323,17 @@
-    return stateObj.attributes.target_temp_low ||
-        stateObj.attributes.target_temp_high;
+  supportsTemperatureRange(stateObj) {
+    return (stateObj.attributes.supported_features & 4) !== 0;
```
To preserve previous behavior of (high or low) compare with "6"

ciotlosm replied:
Will fix that next, but I'm still unsure if using supported_features is correct because it breaks all climate demo components display.

balloob commented:
I would not be surprised if the demo platform has not been kept up to date.

ciotlosm commented:
@balloob would it be ok the PR if I update demo platform as well and assume rest of Thermostats just use correctly supported_features?

ciotlosm committed:
Added more supported_features and fixed fan mode bug

ciotlosm commented on src/more-infos/more-info-climate.html:
```diff
@@ -309,7 +325,7 @@
     if (this.stateObj.attributes.fan_list) {
       this.fanIndex =
-          this.stateObj.attributes.fan_list.indexOf(this.stateObj.attributes.fan_list);
+          this.stateObj.attributes.fan_list.indexOf(this.stateObj.attributes.fan_mode);
```
This was a bug - Should I raise an issue and link it in the PR description as well?

ciotlosm replied:
Raised #852 to make sure it's tracked.

ciotlosm commented:
@andrey-git I've updated more controls to use supported_features including isOn. Whenever you have the time to have a look and let me know if anything else is required. I've updated demo platform that was already merged in backend.

schotime commented:
Can't wait for this. Been bugging me all week. Thought there was a problem with my component.

uchagani commented:
I'm still seeing redundant fields on the climate card with version 0.65.4.  I'm in `Heat` mode but the `auto` temperature adjustments are shown.

![image](https://user-images.githubusercontent.com/1053584/37377876-e4e58c72-2702-11e8-9232-5190d5f00d9b.png)

ciotlosm commented:
@uchagani Please don't comment on closed pull requests. In your case the problem was not supposed to be fixed by this specific PR. It requires a new PR where extra conditions are applied to avoid showing those empty fields.
